chromium_src/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc

```diff
@@ -52,13 +52,40 @@ String PluginReplacementName(FarblingPRNG* prng) {
   return result.ToString();
 }
 
-void FarblePlugins(DOMPluginArray* owner,
-                   PluginData* data,
+// General helper to add a mime type to a plugin info, with a single extension
+MimeClassInfo* AddMimeTypeToPluginInfo(PluginInfo* plugin_info,
+                                       const String& mime_type,
+                                       const String& description,
+                                       const char* extension) {
+  blink::Vector<String> extensions{String(extension)};
+  auto* mime_info = MakeGarbageCollected<MimeClassInfo>(
+      mime_type, description, *plugin_info, extensions);
+  plugin_info->AddMimeType(mime_info);
+  return mime_info;
+}
+
+// Helper to create a fake plugin with multiple mime types
+DOMPlugin* BraveMakeFakePlugin(blink::LocalDOMWindow* window,
+                               const String& name,
+                               const String& filename,
+                               const String& description,
+                               const std::vector<const char*>& mime_types,
```

Comment:
can that be a span of string_view?

Reply:
Actually I just tested this and we can't because the upstream code is taking `const blink::String&` and `blink::StringView` is still to recent, so it will be a while before it becomes more common everywhere.